vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -336,6 +336,74 @@ func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isRead
 	return mountpoint, fs.MountWithID(mountpoint, fstype, id, isReadOnly)
 }
 
+// private function that does the job of mounting volume in conjunction with refcounting
+// Called under a lock if refcounting hasn't been successful, else without a lock
+// Don't do refcounting, just the mount if refcounting hasn't been successful.
+func (d *VolumeDriver) processMount(r volume.MountRequest, doRefcnt bool) volume.Response {
+	// If the volume is already mounted , just increase the refcount.
+	//
+	// Note: We are deliberately incrementing refcount first, before trying
+	// to do anything else. If Mount fails, Docker will send Unmount request,
+	// and we will happily decrement the refcount there, and will fail the unmount
+	// since the volume will have been never mounted.
+	// Note: for new keys, GO maps return zero value, so no need for if_exists.
+
+	if doRefcnt {
+		refcnt := d.incrRefCount(r.Name) // save map traversal
+		log.Debugf("volume name=%s refcnt=%d", r.Name, refcnt)
+		if refcnt > 1 {
+			log.WithFields(
+				log.Fields{"name": r.Name, "refcount": refcnt},
+			).Info("Already mounted, skipping mount. ")
+			return volume.Response{Mountpoint: d.getMountPoint(r.Name)}
+		}
+	}
+
+	// This is the first time we are asked to mount the volume. The disk
+	// ID isn't always cached by the plugin on a host, the disk could have
+	// been created on a different host and getting mounted on another host.
+	// So on the first mount fetch the disk from Photon and add it's ID to
```

Comment:
This looks like old comment. Id never gets added to refmap. Can be removed as part of #1212